DAT257-14-Dashboard

Flaskhals68 opened the pull request:
This will add the base structure for the dashboard.
Cards have been added with place holder elements.

Flaskhals68 committed:
Create DashBoard component

Flaskhals68 committed:
Merge branch 'main' into DAT257-14-Dashboard

Flaskhals68 committed:
Make DashBoard make api calls

erbergli committed:
Add AppBar

Flaskhals68 committed:
Move Api calls to own class

Flaskhals68 committed:
Add table of contents

Flaskhals68 committed:
Remove comments

Flaskhals68 committed:
Add labels on cards

erbergli committed:
Merge branch 'DAT257-14-Dashboard' into DAT257-14-Dashboard-AppBar

Flaskhals68 committed:
Remove unused css

Flaskhals68 committed:
Save api url to string

This will make api calls more readable.

Flaskhals68 committed:
Delete DashBoard.css

Flaskhals68 committed:
Update Client.sln

Flaskhals68 committed:
Restore revert changes

filipkauffeldt commented on Client/API/ApiRequestHandler.cs:
```diff
@@ -0,0 +1,55 @@
+            }
+        }
+
+        public static async Task<Country> FetchCountry(int id)
```
It should be a string with the country ISO code to the database. The endpoint is not up to date with that yet, but should also be changed

filipkauffeldt commented on Client/API/Country.cs:
```diff
@@ -0,0 +1,14 @@
+        public string? Continent { get; init; }
+        public string? Description { get; init; }
+        public Collection<Data>? Data { get; init; }
+
```
Remove new line

filipkauffeldt commented on Client/Layout/MainLayout.razor:
```diff
@@ -5,8 +5,11 @@
 
     <main>
-        <div class="top-row px-4">
+@*         <div class="top-row px-4">
```
Remove unused

filipkauffeldt commented:
Should the APIRequestHandler maybe be injected as a dependency in the program.cs file and have a http client injected? I think that is best practice, and also allows for testing the methods by mocking the http client

Flaskhals68 committed:
Implement dependency injection

This will use dependency injection for the ApiRequestHandler instead of using a static class.

Flaskhals68 committed:
Remove unused html element

Flaskhals68 committed:
Update api endpoint

filipkauffeldt reviewed (commented):
Looks good, but should the api request handler not be tested?

Flaskhals68 committed:
Merge branch 'main' into DAT257-14-Dashboard

Flaskhals68 committed:
Add tests for ApiRequestHandler

Flaskhals68 committed:
Update API contracts

jakobschy reviewed (dismissed):
Looks good

Flaskhals68 committed:
Update fetched type

erbergli reviewed (approved):
LGTM!!!!

Huffla reviewed (approved):
TYes
